.github/workflows/qa.yml

```
@@ -0,0 +1,47 @@
+name: Run QA Framework tests on Github Actions
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    services:
+      firefox:
+        image: selenium/node-firefox:3.141.59-gold
+        env:
+          HUB_HOST: hub
+          HUB_PORT: 4444
+    steps:
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Set up Maven
+        uses: aahmed-se/setup-maven@v3
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+        with:
+          maven-version: 3.6.1
+      - name: Checkout ui test framework
+        uses: actions/checkout@master
+        with:
+          repository: openmrs/openmrs-contrib-uitestframework
+      - name: Build ui test framework
+        run: mvn clean install
+      - name: Checkout distro/uitests
+        uses: actions/checkout@master
+        with:
+          repository: openmrs/openmrs-distro-referenceapplication
+      - name: Build distro/uitests
+        run: mvn clean install
+      - name: Checkout qaframework
+        uses: actions/checkout@master
+        with:
+          repository: openmrs/openmrs-contrib-qaframework
+      - name: Run qaframework on selenium
+        run: |
+          echo "cucumber.publish.enabled=true" > src/test/resources/cucumber.properties
+          cp -f src/test/resources/org/openmrs/uitestframework/test-travis-ci.properties src/test/resources/org/openmrs/uitestframework/test.properties
+          mvn clean install
```

```suggestion
          mvn -B clean install
```

`-B` enables batch-mode, which is useful when building on CI (it doesn't stop on errors, among other things). Note, too that Travis's default Maven build is actually:

```
mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
mvn test -B
```

If you want to replicate the flow exactly (hence why I had [two steps](https://gist.github.com/ibacher/d8ad221092be465dcef9a4ec6434c012#file-travis-gh-sh-L44) in my example).

---

i like option -B and will have it in, does github actions cache and parhaps i need to manage the cache in the config too!

---

Caching the Maven repo needs a different action, but I've used [this one](https://gist.github.com/ibacher/56d7535b398b9716b886a27eaed7e62d#file-maven-yml-L29) before with some success.